### PR TITLE
Iterators task 1 kosiakova

### PR DESCRIPTION
**Pull request by olsya**
Given an input string, reverse the string word by word

### CODE REVIEW

File: iterators_task_1/student/src/impl.cpp [hunk 1]
```diff
@@ -1 +1,22 @@
...
+#include <vector>
+#include <algorithm>
+#include <iostream>
+
+std::string reverseWords(const std::string& input)
+{
+    using ii_string = std::istream_iterator<std::string>;
+    using oi_string = std::ostream_iterator<std::string>;
+
+    std::istringstream iss(input);
```
Comment (smay1613) [inline iterators_task_1/student/src/impl.cpp:12]: _please follow the same style of initialization ({})_

File: iterators_task_1/student/src/impl.cpp [hunk 2]
```diff
@@ -1 +1,22 @@
...
+    std::vector<std::string> words{ii_string{iss}, ii_string{}};
+
+    if (words.empty()) {
+        return "";
+    }
+
+    std::stringstream result;
+    std::copy(words.rbegin(), words.rend(), oi_string(result, " "));
+    return result.str().erase(result.str().size() - 1);
```
Comment (smay1613) [inline iterators_task_1/student/src/impl.cpp:21]: prefer pop_back()

File: iterators_task_1/student/src/impl.cpp (hunk 1 above)
Comment (olsya) [reply iterators_task_1/student/src/impl.cpp:12]: Done

File: iterators_task_1/student/src/impl.cpp (hunk 2 above)
Comment (olsya) [reply iterators_task_1/student/src/impl.cpp:21]: Done

**Pull request closed without merging**